src/index.jsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from 'react';
+import ReactDOM from 'react-dom';
+
+/* function Condition({ children }) { // (5) Condition컴포넌트: (4)으로부터 props의 children을 받고, children으로 조건문실행
+    if (children === 1) {
+        return (
+            console.log(children)
+        )
+    }
+    if (children === 2) {
+        return (
+            console.log(children)
+        )
+    }
+    if (children === 3) {
+        return (
+            console.log(3)
+        )
+    }
+    if (children === 4) {
+        return (
+            console.log(4)
+        )
+    }
+    if (children === 5) {
+        return (
+            console.log(5)
+        )
+    }
+}*/
+
+function Button({ children, onClick }) { // (4) Button컴포넌트: App에서 실행됨, (3)으로부터 i를 children으로 받음
+    return (
+        <button type="button" onClick={() => onClick(children)}>
+            {children}
+        </button>
+    )
+}
+
+function Numeric() { // (3) Numeric컴포넌트
+    return (
+        [1, 2, 3, 4, 5].map((i) => (
+            <Button>
+                {i}
+            </Button>
+        ))
+    )
+}
```

Comment:
```suggestion
function Button({ children, onClick }) {
    return (
        <button type="button" onClick={onClick}>
            {children}
        </button>
    )
}

function NumberButtons({ onClick }) { 
    return (
        [1, 2, 3, 4, 5].map((i) => (
            <Button onClick={() => onClick(i)}>
                {i}
            </Button>
        ))
    )
}
```

먼저 버튼 컴포넌트의 이름을 의미가 명확하게 드러나게 바꿔주시면 좋을 것 같아요.
그리고 버튼 컴포넌트는 범용으로 사용할 수 있는 구조를 만들기 위해 이벤트 핸들러 함수를 그대로 전달받는게 좋아요.
대신 매개변수를 전달하는 구조는 상위 컴포넌트에서 처리해줘야하죠.

기존에 작성하신 코드와 리뷰를 참고하셔서 제가 작성한 코드를 비교해보시면 도움이 많이 되실거에요 ;)

Reply:
코멘트 감사드려요 😃

위의 Button, NumberButtons 컴포넌트로 예를 들자면,
NumberButtons의 1,2,3,4,5가 i에 담겨서 인자로써 onClick(이벤트 핸들러 함수)에 전달되고,
또, Button컴포넌트의 props로써 children매개변수로도 전달되어서
결국 Button이라는 상위 컴포넌트에서 children매개변수를 App컴포넌트로 전달해주는 구조를 취하게 되는 것인가요?

제 이해가 맞는건지 궁금합니다... 😅

Reply:
> 위의 Button, NumberButtons 컴포넌트로 예를 들자면, NumberButtons가 Buttons의 상위 컴포넌트가 되는 것이지요?

네 NumberButtons 에서 Button 을 호출하기 것이므로 NumberButtons 가 상위컴포넌트라고 보시면 돼요.

> 관심사의 분리관점으로 본다면,
Button컴포넌트가 화면에 그저 그려주는 역할인 Counter,
NumberButtons컴포넌트가 상태를 관리해주는 역할인 App인 것이구요?

Button 과 NumberButtons 컴포넌트 모두 UI 를 그려주는 presentational 컴포넌트일뿐 상태관리와 UI 관심사로 나눠지는 건 아니에요!

> NumberButtons의 1,2,3,4,5가 i에 담겨서 인자로써 onClick(이벤트 핸들러 함수)에 전달되고,
또, Button컴포넌트의 props로써 children매개변수로도 전달되어서
결국 Button컴포넌트에서 children매개변수를 App컴포넌트로 전달해주는 구조를 취하게 되는 것인가요?

이 부분은 [이벤트 버블링과 캡처링](https://ko.javascript.info/bubbling-and-capturing)을 공부해보시면 좋을 것 같아요.

Reply:
확인하였습니다! 👏
감사합니당!